http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java

```diff
@@ -117,68 +135,129 @@ public static FeignClientFactory of(
-        return createProxy(sslSocketFactory, uri, type, options);
+    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, String uri, Class<T> type,
+            ProxyConfigurationProvider proxyConfigurationProvider) {
+        return createProxy(sslSocketFactory, uri, type, proxyConfigurationProvider, options);
+    }
+
+    /**
+     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and {@link
+     * okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}.
+     */
+    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, String uri, Class<T> type,
+            ProxyConfigurationProvider proxyConfigurationProvider, Request.Options requestOptions) {
+        return createProxy(sslSocketFactory, ImmutableSet.of(uri), type, proxyConfigurationProvider, requestOptions);
     }
 
+    /**
+     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and
+     * {@link okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}. A {@link FailoverFeignTarget}
+     * is used to cycle through the given uris on failure if there's more than one uri.
+     */
+    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, Set<String> uris, Class<T> type) {
+        return createProxy(sslSocketFactory, uris, type, options);
+    }
 
     /**
-     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and feign {@link
-     * feign.Client.Default} HTTP client. A {@link FailoverFeignTarget} is used to cycle through the given uris on
-     * failure.
+     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, and {@link
+     * okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}. A {@link FailoverFeignTarget} is used
+     * to cycle through the given uris on failure if there's more than one uri.
      */
     public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, Set<String> uris, Class<T> type,
             Request.Options requestOptions) {
-        FailoverFeignTarget<T> target = new FailoverFeignTarget<>(uris, type, backoffStrategy);
-        Client client = clientSupplier.createClient(sslSocketFactory, userAgent);
-        client = target.wrapClient(client);
-        return Feign.builder()
+        return createProxy(sslSocketFactory, uris, type, defaultProxyConfigurationProvider, requestOptions);
+    }
+
+    /**
+     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, proxy
+     * configuration provider, and {@link okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}.
+     * A {@link FailoverFeignTarget} is used to cycle through the given uris on failure if there's more than one uri.
+     */
+    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, Set<String> uris, Class<T> type,
```

Comment:
I will propose a new API in the next couple of hours, shooting for an RC
release with proxy support by EOD.

On Tue, Jul 12, 2016, 10:38 PM Mark Elliot <notifications@github.com> wrote:

> In
> http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java
> <https://github.com/palantir/http-remoting/pull/158#discussion_r70557048>:
>
> >       */
> >      public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, Set<String> uris, Class<T> type,
> >              Request.Options requestOptions) {
> > -        FailoverFeignTarget<T> target = new FailoverFeignTarget<>(uris, type, backoffStrategy);
> > -        Client client = clientSupplier.createClient(sslSocketFactory, userAgent);
> > -        client = target.wrapClient(client);
> > -        return Feign.builder()
> > +        return createProxy(sslSocketFactory, uris, type, defaultProxyConfigurationProvider, requestOptions);
> > +    }
> > +
> > +    /**
> > +     * Constructs a dynamic proxy for the specified type, using the supplied SSL factory if is present, proxy
> > +     * configuration provider, and {@link okhttp3.OkHttpClient} wrapped in {@link feign.okhttp.OkHttpClient}.
> > +     * A {@link FailoverFeignTarget} is used to cycle through the given uris on failure if there's more than one uri.
> > +     */
> > +    public <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactory, Set<String> uris, Class<T> type,
>
> Rob and I have been discussing at least hiding Feign from the API in this
> regardless. Conjure gives us optionality for change, but isn't a golden
> ticket. Certainly things are simpler if we hide the impl details, though.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/158/files/06fb4d86888244f9936355100bd9c72b92e333ba#r70557048>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGOdwZonpx6wg7qVBjJg8m8pj5vZ8frtks5qVE-EgaJpZM4JJ5TO>
> .
>